inst/include/population_dynamics/recruitment/functors/recruitment_base.hpp

```diff
@@ -18,106 +18,157 @@
-    Type recruit_bias_adjustment_size = this->recruit_bias_adjustment.size();
-
-    if (!this->use_recruit_bias_adjustment) {
-      for (size_t i = 0; i < recruit_bias_adjustment_size; i++) {
-        this->recruit_bias_adjustment_fraction[i] = 1.0;
-        this->recruit_bias_adjustment[i] = 0.0;
-      }
-    } else {
-      for (size_t i = 0; i < recruit_bias_adjustment_size; i++) {
-        // Initially fixing bias adjustment (b_y in collobarative
-        // workflow specification) to 1.0.
-        // In the future, this would be set by the user.
-        this->recruit_bias_adjustment_fraction[i] = 1.0;
-        this->recruit_bias_adjustment[i] =
-            0.5 * fims::exp(this->log_sigma_recruit) *
-            fims::exp(this->log_sigma_recruit) *
-            this->recruit_bias_adjustment_fraction[i];
-      }
-    }
-  }
-};
-
-template <class Type>
-uint32_t RecruitmentBase<Type>::id_g = 0;
-}  // namespace fims
+        /** @brief Constructor.
+         */
+        RecruitmentBase() {
+            this->id = RecruitmentBase::id_g++;
+        }
+
+        virtual ~RecruitmentBase() {
+        }
+
+/**
+ * @brief Prepares the recruitment bias adjustment vector.
+ * 
+ */
+        void Prepare() {
+          this->recruit_bias_adjustment_fraction.resize(this->recruit_deviations.size());
+          this->recruit_bias_adjustment.resize(this->recruit_deviations.size());
+            this->PrepareConstrainedDeviations();
+//            this->PrepareBiasAdjustment();
+//            if (this->use_recruit_bias_adjustment) {
+//                for (size_t i = 0; i < this->recruit_deviations.size(); i++) {
+//                    this->recruit_bias_adjustment[i] = -0.5 * fims::exp(this->log_sigma_recruit) * fims::exp(this->log_sigma_recruit);
+//
+//                }
+//            } else {
+//                for (size_t i = 0; i < this->recruit_deviations.size(); i++) {
+//                    this->recruit_bias_adjustment[i] = 0.0;
+//                }
+//            }
+        }
+
+        /** @brief Calculates the expected recruitment for a given spawning input.
+         *
+         * @param spawners A measure for spawning output.
+         * @param ssbzero A measure for spawning output in unfished population.
+         *
+         */
+        virtual const Type evaluate(
+                const Type &spawners,
+                const Type &ssbzero) = 0; // need to add input parameter values
+
+        virtual const Type evaluate_nll() {
+
+            Type nll = 0.0; /*!< The negative log likelihood value */
+
+            if (!this->estimate_recruit_deviations) {
+                return nll;
+            } else {
+                #ifdef TMB_MODEL
+                fims::Dnorm<Type> dnorm;
+                dnorm.sd = fims::exp(this->log_sigma_recruit);
+                for (size_t i = 0; i < this->recruit_deviations.size(); i++) {
+                    dnorm.x = fims::log(this->recruit_deviations[i]);
+                    dnorm.mean = 0.0;
```

Comment:
question: we're not clear - should `recruitment_deviations` be entered in log space rather than being logged only within the likelihood?

Reply:
recruit deviations are in linear space when [input](https://github.com/NOAA-FIMS/FIMS/blob/10d5103cecdf2e37b724eb15dbf66dfb146c472b/tests/testthat/test-fims-estimation.R#L46) into the model and when used in [CalculateRecruitment](https://github.com/NOAA-FIMS/FIMS/blob/5cb12ac9209daf94f955895db5299e7cbce91517/inst/include/population_dynamics/population/population.hpp#L367). If we update rec devs to be in logspace, we will need to update code in these two locations. I personally find it hard to keep track of what is supposed to be in log versus linear space and would prefer renaming `recruitment_deviations` to `log_recruitment_deviations` or some variety of `logR` if we want to convert this to log space.

Reply:
I think "deviations", either in variable names or plain language, should refer to values that are distributed on either side of 0. At least I would find that much less confusing.

If it's important to maintain non-log values for annual recruitment values, I would call them something like `recruitment_scalars` or `recruitment_multipliers`.

Reply:
This has been added as issue#402